namer/rancher/src/main/scala/io/buoyant/namer/rancher/RancherClient.scala

```diff
@@ -0,0 +1,140 @@
+package io.buoyant.namer.rancher
+
+import com.fasterxml.jackson.databind.{ObjectMapper, PropertyNamingStrategy}
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+
+import com.twitter.finagle.{Service, Http, Address, Stack}
+import com.twitter.finagle.http
+import com.twitter.finagle.service.Retries
+import com.twitter.finagle.tracing.NullTracer
+import com.twitter.finagle.stats.{Stat, StatsReceiver}
+
+import com.twitter.logging.Logger
+import com.twitter.util.Activity.State
+import com.twitter.util.{Timer, Var, Activity, Duration, Future, Closable, Return, Throw}
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+case class PortMapping(
+  val ip: String,
+  val publicPort: Int,
+  val privatePort: Int,
+  val protocol: String
+) {
+  // This is a bit of a hack - and should have more error-checking. But it
+  // makes the JSON parsing work.
+  // If anyone knows how to do a better constructor - or JSON Deserializer,
+  // that would be nice
+  def this(mapping: String) = this(
+    mapping.split(":")(0),
+    mapping.split(":")(1).toInt,
+    mapping.split(":")(2).split("/")(0).toInt,
+    mapping.split(":")(2).split("/")(1)
+  )
+  override def toString(): String = s"$ip:$privatePort:$publicPort/$protocol"
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+case class RancherContainer(
+  createIndex: Int,
+  dns: List[String],
+  ips: List[String],
+  primaryIp: String,
+  labels: Map[String, String],
+  name: String,
+  ports: List[PortMapping],
+  serviceName: String,
+  stackName: String,
+  state: String
+) {
+  def toAddr(port: Int): Option[Address] = state match {
+    case "running" => Some(Address(primaryIp, port))
+    case _ => None
+  }
+}
+
+class RancherClient(
+  log: Logger,
+  params: Stack.Params,
+  stats: StatsReceiver
+)(implicit val timer: Timer) {
+  private[this] val objectMapper = new ObjectMapper() with ScalaObjectMapper
+  objectMapper.registerModule(DefaultScalaModule)
+  objectMapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE)
+
+  private[this] val numVersions = stats.counter("metadata_versions")
+
+  private[this] val maxWait = 30
```

Comment:
Probably a good idea. Since the long-polling API isn't even documented from Rancher's side, I have no idea how long of a wait it actually supports - so making it configurable makes a lot of sense. I'll do that.